tests/backup/common.go

```diff
@@ -0,0 +1,1306 @@
+		Cluster:          clusterName,
+		NamespaceMapping: namespaceMapping,
+		BackupRef: &api.ObjectRef{
+			Name: backupName,
+			Uid:  bkpUid,
+		},
+	}
+	_, err := backupDriver.CreateRestore(ctx, createRestoreReq)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateRestoreGetErr creates restore
+func CreateRestoreGetErr(restoreName string, backupName string,
+	namespaceMapping map[string]string, clusterName string, orgID string) (err error) {
+	backupDriver := Inst().Backup
+	createRestoreReq := &api.RestoreCreateRequest{
+		CreateMetadata: &api.CreateMetadata{
+			Name:  restoreName,
+			OrgId: orgID,
+		},
+		Backup:           backupName,
+		Cluster:          clusterName,
+		NamespaceMapping: namespaceMapping,
+	}
+	//ctx, err := backup.GetPxCentralAdminCtx()
+	ctx, err := backup.GetAdminCtxFromSecret()
+	if err != nil {
+		return err
+	}
+	_, err = backupDriver.CreateRestore(ctx, createRestoreReq)
+	if err != nil {
+		return err
+	}
+
+	// TODO: validate createClusterResponse also
+
+	return err
+}
+
+func getBackupUID(backupName, orgID string) (string, error) {
+	backupDriver := Inst().Backup
+	ctx, err := backup.GetAdminCtxFromSecret()
+	if err != nil {
+		return "", err
+	}
+	backupUID, err := backupDriver.GetBackupUID(ctx, backupName, orgID)
+	if err != nil {
+		return "", err
+	}
+	return backupUID, nil
+}
+
+func getSizeOfMountPoint(podName string, namespace string, kubeConfigFile string) (int, error) {
+	var number int
+	ret, err := kubectlExec([]string{podName, "-n", namespace, "--kubeconfig=", kubeConfigFile, " -- /bin/df"})
+	if err != nil {
+		return 0, err
+	}
+	for _, line := range strings.SplitAfter(ret, "\n") {
+		if strings.Contains(line, "pxd") {
+			ret = strings.Fields(line)[3]
+		}
+	}
+	number, err = strconv.Atoi(ret)
+	if err != nil {
+		return 0, err
+	}
+	return number, nil
+}
+
+func kubectlExec(arguments []string) (string, error) {
+	if len(arguments) == 0 {
+		return "", fmt.Errorf("no arguments supplied for kubectl command")
+	}
+	cmd := exec.Command("kubectl exec -it", arguments...)
+	output, err := cmd.Output()
+	log.Debugf("command output for '%s': %s", cmd.String(), string(output))
+	if err != nil {
+		return "", fmt.Errorf("error on executing kubectl command, Err: %+v", err)
+	}
+	return string(output), err
+}
+
+func createUsers(numberOfUsers int) []string {
+	users := make([]string, 0)
+	log.InfoD("Creating %d users", numberOfUsers)
+	var wg sync.WaitGroup
+	for i := 1; i <= numberOfUsers; i++ {
+		userName := fmt.Sprintf("testuser%v-%v", i, time.Now().Unix())
+		firstName := fmt.Sprintf("FirstName%v", i)
+		lastName := fmt.Sprintf("LastName%v", i)
+		email := fmt.Sprintf("%v@cnbu.com", userName)
+		wg.Add(1)
+		go func(userName, firstName, lastName, email string) {
+			err := backup.AddUser(userName, firstName, lastName, email, password)
+			Inst().Dash.VerifyFatal(err, nil, fmt.Sprintf("Creating user - %s", userName))
+			users = append(users, userName)
+			wg.Done()
```

Comment:
make defer wg.Done and move to first line

Reply:
also add defer ginkgorecover

Reply:
done